PowerPointLabs/PowerPointLabs/Utils/ShapeUtil.cs

```diff
@@ -354,6 +361,35 @@ public static ShapeRange CorruptionCorrection(ShapeRange shapes, PowerPointSlide
             return ownerSlide.ToShapeRange(correctedShapeList);
         }
 
+        /// <summary>
+        /// To avoid corrupted shape.
+        /// Corrupted shape is produced when delete or cut a shape programmatically, but then users undo it.
+        /// After that, most of operations on corrupted shapes will throw an exception.
+        /// One solution for this is to re-allocate its memory: simply cut/copy and paste.
+        /// </summary>
+        /// <param name="shape"></param>
+        /// <returns></returns>
+        public static Shape CutPasteShape(Shape shape)
+        {
+            // Save clipboard onto a temp slide, this is similar to code in PasteLabActionHandler.cs
+            // Have no choice to use deprecated methods because ShapeUtil does not use ActionFramework
+            PowerPointPresentation presentation = PowerPointPresentation.Current;
+
+            PowerPointSlide tempClipboardSlide = presentation.AddSlide();
+            ShapeRange tempClipboardShapes = ClipboardUtil.PasteShapesFromClipboard(tempClipboardSlide);
+
+            shape.Cut();
+            shape = PowerPointCurrentPresentationInfo.CurrentSlide.Shapes.Paste()[1];
+
+            // Revert clipboard. Note that cannot be reverted if last copied item was a placeholder
+            if (tempClipboardShapes != null)
+            {
+                tempClipboardShapes.Copy();
+            }
+            tempClipboardSlide.Delete();
```

Comment:
Duplicate codes with ConvertToPicture.cs?

Reply:
Have shifted duplicate codes to ClipboardUtil.cs to facilitate reuse. Thanks, will request a peer review first next time.